fdbserver/workloads/SnapTest.actor.cpp

```diff
@@ -0,0 +1,618 @@
+			it = messages.erase(it);
+		} else {
+			++it;
+		}
+	}
+	return;
+}
+
+void printMessages(std::vector<Future<TraceEventFields>>& messages) {
+	for (int i = 0; i < messages.size(); i++) {
+		TraceEvent("SnapTestMessages").detail("I", i).detail("Value", messages[i].get().toString());
+	}
+	return;
+}
+
+struct SnapTestWorkload : TestWorkload {
+public: // variables
+	int numSnaps; // num of snapshots to be taken
+	              // FIXME: currently validation works on numSnap = 1
+	double maxSnapDelay; // max delay before which a snapshot will be taken
+	bool snapCheck; // check for the successful snap create
+	int testID; // test id
+	UID snapUID; // UID used for snap name
+	std::string restartInfoLocation; // file location to store the snap restore info
+	int maxRetryCntToRetrieveMessage; // number of retires to do trackLatest
+
+public: // ctor & dtor
+	SnapTestWorkload(WorkloadContext const& wcx)
+	  : TestWorkload(wcx), numSnaps(0), maxSnapDelay(0.0), snapCheck(false), testID(0), snapUID() {
+		TraceEvent("SnapTestWorkload Constructor");
+		std::string workloadName = "SnapTest";
+		maxRetryCntToRetrieveMessage = 10;
+
+		numSnaps = getOption(options, LiteralStringRef("numSnaps"), 0);
+		maxSnapDelay = getOption(options, LiteralStringRef("maxSnapDelay"), 25.0);
+		snapCheck = getOption(options, LiteralStringRef("snapCheck"), false);
+		testID = getOption(options, LiteralStringRef("testID"), 0);
+		restartInfoLocation =
+		    getOption(options, LiteralStringRef("restartInfoLocation"), LiteralStringRef("simfdb/restartInfo.ini"))
+		        .toString();
+	}
+
+public: // workload functions
+	std::string description() override { return "SnapTest"; }
+	Future<Void> setup(Database const& cx) override {
+		TraceEvent("SnapTestWorkloadSetup");
+		return Void();
+	}
+	Future<Void> start(Database const& cx) override {
+		TraceEvent("SnapTestWorkloadStart");
+		if (clientId == 0) {
+			return _start(cx, this);
+		}
+		return Void();
+	}
+
+	Future<bool> check(Database const& cx) override {
+		TraceEvent("SnapTestWorkloadCheck").detail("ClientID", clientId).detail("SnapCheck", this->snapCheck);
+		if (!this->snapCheck || clientId != 0) {
+			TraceEvent("SnapTestCheckSucc");
+			return true;
+		}
+		switch (this->testID) {
+		case 0:
+		case 1:
+		case 2:
+		case 3: {
+			Future<std::vector<WorkerInterface>> proxyIfaces;
+			return (verifyExecTraceVersion(cx, this));
+			break;
+		}
+		case 4: {
+			std::string token = "DisableTLogPopTimedOut";
+			return verifyTLogTrackLatest(cx, this, token);
+			break;
+		}
+		case 5: {
+			std::string token = "TLogPopDisableEnableUidMismatch";
+			return verifyTLogTrackLatest(cx, this, token);
+			break;
+		}
+		case 6: {
+			std::string token = "SnapFailIgnorePopNotSet";
+			return verifyTLogTrackLatest(cx, this, token);
+			break;
+		}
+		case 7: {
+			std::string token = "SnapFailedDisableTLogUidMismatch";
+			return verifyTLogTrackLatest(cx, this, token);
+			break;
+		}
```

Comment:
UID in the snapshot acts as the unique identifier, snapshot create binary/scripts gets the UID and can use it as name or any other method to prevent multiple snapshots getting created.